x-pack/plugins/security_solution/public/common/components/dashboards/dashboards_table.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { Search } from '@elastic/eui';
+import { EuiInMemoryTable } from '@elastic/eui';
+import {
+  useSecurityDashboardsTableItems,
+  useDashboardsTableColumns,
+} from '../../containers/dashboards/use_security_dashboards';
+
+const DASHBOARDS_TABLE_SEARCH: Search = {
+  box: {
+    incremental: true,
+  },
+} as const;
+
+export const DashboardsTable: React.FC = () => {
+  const items = useSecurityDashboardsTableItems();
+  const columns = useDashboardsTableColumns();
+
+  return (
+    <EuiInMemoryTable
```

Comment:
We are loading all dashboards without pagination. Would it be a problem if the user has a gazillion dashboards?

Reply:
Yup, good point, I have been thinking about it as well. 
My reasoning is that those "security" dashboards will be created manually, I think it is very unlikely that we have to load an astronomical number of "security" dashboards that could cause a hit in performance.

Implementing pagination to this table would mean to add quite a lot of complexity to the search field logic and to the savedObjects querying.

IMHO, I prefer not to pre-solve performance issues that are unlikely to happen, and keep the code simple. If someday we detect performance degradations we can implement pagination then.

@YulNaumenko what do you think?

Reply:
I think for now we can go without the pagination, because I don't think that adding this complexity will be valuable with the current product expectations.